### PR TITLE
User management page

### PR DESCRIPTION
**Pull request by AxelHumbert**

### CODE REVIEW

File: src/front/src/conf/userController.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    page: number = 0,
+    filter?: {filterField: string, filterValue: string },
+    sort?: { sortField: string, sortOrder: 'asc' | 'desc' }): Promise<PageRequest<User> | undefined> {
+
+    let request = await fetch(window.location.origin + `/api/user/search?size=${size}&page=${page}`)
+    if (request.ok) {
+        let content = await request.json()
+        let user = (content as PageRequest<User>)
+        if (user != null) {
+            console.log(user)
```
Comment (Adrien-COSTANDI) [inline src/front/src/conf/userController.ts]: remove console.log

File: src/front/src/pages/UserTable.tsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+import { InfinityScrollItemsTable, InfinityScrollItemsTableProps, PageRequest, TableColumnDefinition } from "./DisplayTable";
+import { Role, searchUser, User } from "../conf/userController";
+import { englishRoleToFrench } from "../NavBar";
+
+const PAGE_SIZE = 10; // Max items displayed in the user table
+
+const userTableColumns: TableColumnDefinition[] = [
+    {
+        title: "Nom",
+        /*
```
Comment (Adrien-COSTANDI) [inline src/front/src/pages/UserTable.tsx]: remove useless comment

File: src/front/src/pages/UserTable.tsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
+import { Box, Grid, ListItemButton, MenuItem, Select, SelectChangeEvent, Typography } from "@mui/material";
+import React, { useEffect, useState } from "react";
+import { InfinityScrollItemsTable, InfinityScrollItemsTableProps, PageRequest, TableColumnDefinition } from "./DisplayTable";
+import { Role, searchUser, User } from "../conf/userController";
+import { englishRoleToFrench } from "../NavBar";
+
+const PAGE_SIZE = 10; // Max items displayed in the user table
```
Comment (Adrien-COSTANDI) [inline src/front/src/pages/UserTable.tsx]: set to 30

File: src/front/src/pages/UserTable.tsx [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        */
+    },
+    {
+        title: "Prénom",
+    },
+    {
+        title: "Rôle",
+    },
+    {
+        title: "",
```
Comment (Adrien-COSTANDI) [inline src/front/src/pages/UserTable.tsx]: add a comment to explain why the title is empty

File: src/front/src/pages/UserTable.tsx [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+            setHasMore(result.total > PAGE_SIZE)
+        });
+    }, [])
+
+    useEffect(() => {
+        const fetchRoleList = async () => {
+            const response = await fetch('/api/user/allRoles');
+            const data = await response.json();
+            setUserRoleList(data);
+            console.log(data);
```
Comment (Adrien-COSTANDI) [inline src/front/src/pages/UserTable.tsx]: remove console.log

File: src/front/src/pages/UserTable.tsx [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+        }
+        fetchRoleList();
+    }, []);
+
+    useEffect(() => {
+        const fetchRoleList = async () => {
```
Comment (Adrien-COSTANDI) [inline src/front/src/pages/UserTable.tsx]: rename variable

File: src/front/src/pages/UserTable.tsx [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+                    return;
+                }
+                let updatedUser = tableData.find(u => u.id === user.id)
+                if (!updatedUser) {
+                    console.log("User id for the update notification is not found.")
+                    return
+                }
+                // Update role of user
+                user.role = role
+                setTableData([...tableData])
```
Comment (Adrien-COSTANDI) [inline src/front/src/pages/UserTable.tsx:91]: what is this ?

File: src/main/java/fr/uge/chargepointconfiguration/user/UserController.java [hunk 8]
```diff
@@ -188,4 +189,22 @@ public PageDto<UserDto> getPage(
...
+  @Operation(summary = "Get all availables roles")
+  @ApiResponse(responseCode = "200",
+          description = "Found roles",
+          content = { @Content(mediaType = "application/json",
+                  schema = @Schema(implementation = User.Role.class))
+          })
+  @GetMapping(value = "/allRoles")
+  public List<User.Role> getAllRoles() {
+    return Arrays.stream(User.Role.values())
+            //.filter(r -> r != getAuthenticatedUser().role())
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/user/UserController.java]: remove comment

**Review by Adrien-COSTANDI — CHANGES_REQUESTED** [review]
I will change what is necessary

**Review by Adrien-COSTANDI — APPROVED** [review]

File: src/front/src/pages/UserTable.tsx [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+            body: JSON.stringify({
+                id: user.id,
+                role: role
+            }),
+            headers: {
+                "Content-Type": "application/json"
+            }
+        }).then(response => {
+            if (response.ok) {
+                // TODO : refactor to use the DTO in the response
```
Comment (Adrien-COSTANDI) [inline src/front/src/pages/UserTable.tsx:79]: important TODO

File: src/front/src/pages/UserTable.tsx [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+
+                if (!user) {
+                    return;
+                }
+                let updatedUser = tableData.find(u => u.id === user.id)
+                if (!updatedUser) {
+                    console.log("User id for the update notification is not found.")
```
Comment (tvelu77) [inline src/front/src/pages/UserTable.tsx:86]: Console.log should be removed

File: src/front/src/pages/UserTable.tsx [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+            console.log(event.target.value as string)
+    } 
+
+    let props: InfinityScrollItemsTableProps<User> = {
+        columns: userTableColumns,
+        key: "user-table",
+        data: tableData,
+        hasMore: hasMore,
+        error: error,
+        onSelection: user => { console.log("Selected item : " + user.id) },
```
Comment (tvelu77) [inline src/front/src/pages/UserTable.tsx:103]: Console log should be removed

**Review by tvelu77 — APPROVED** [review]

**Pull request merged**